mcs/class/corlib/System/TimeZoneInfo.Unity.cs

```diff
@@ -124,44 +122,45 @@ static TimeZoneInfo CreateLocalUnity ()
 		{
 			Int64[] data;
 			string[] names;
+			string daylightDisplayName = null;
 			//Some timezones start in DST on january first and disable it during the summer
 			bool dst_inverted;
+			var adjustmentRulesList = new List<AdjustmentRule> ();
+
+			//the icall only supports years from 1970 through 2037.
+			int firstSupportedDate = 1971;
+			int lastSupportedDate = 2037;
 			int currentYear = DateTime.UtcNow.Year;
+
+			// Find the most recent daylight saving time display name by working backwards
+			bool disableDaylightSavings = true;
+			for (var year = lastSupportedDate; year >= firstSupportedDate; year--) {
+				if (!System.CurrentSystemTimeZone.GetTimeZoneData (year, out data, out names, out dst_inverted))
+					throw new NotSupportedException ("Can't get timezone name.");
+
+				disableDaylightSavings = data[(int)TimeZoneData.AdditionalDaylightOffsetIdx] == 0 || data[(int)TimeZoneData.UtcOffsetIdx] == data[(int)TimeZoneData.AdditionalDaylightOffsetIdx];
+				if (!disableDaylightSavings)
+				{
+					daylightDisplayName = names[(int)TimeZoneNames.DaylightNameIdx];
+					break;
+				}
+			}
+
 			if (!System.CurrentSystemTimeZone.GetTimeZoneData (currentYear, out data, out names, out dst_inverted))
 				throw new NotSupportedException ("Can't get timezone name.");
 
 			var utcOffsetTS = TimeSpan.FromTicks (data[(int)TimeZoneData.UtcOffsetIdx]);
 			char utcOffsetSign = (utcOffsetTS >= TimeSpan.Zero) ? '+' : '-';
 			string displayName = "(GMT" + utcOffsetSign + utcOffsetTS.ToString (@"hh\:mm") + ") Local Time";
 			string standardDisplayName = names[(int)TimeZoneNames.StandardNameIdx];
-			string daylightDisplayName = names[(int)TimeZoneNames.DaylightNameIdx];
+			daylightDisplayName ??= names[(int)TimeZoneNames.DaylightNameIdx];
```

Comment:
line 144 already handles the assignment of `daylightDisplayName`, correct?

Reply:
it looks like this is a "if the above doesn't work for some reason then revert to previous behavior change to me?